server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -427,46 +421,13 @@ private static boolean upgradeLegacyNodeFolders(Logger logger, Settings settings
     private void maybeLogPathDetails() throws IOException {
 
         // We do some I/O in here, so skip this if DEBUG/INFO are not enabled:
-        if (logger.isDebugEnabled()) {
-            // Log one line per path.data:
-            StringBuilder sb = new StringBuilder();
-            for (NodePath nodePath : nodePaths) {
-                sb.append('\n').append(" -> ").append(nodePath.path.toAbsolutePath());
-
-                FsInfo.Path fsPath = FsProbe.getFSInfo(nodePath);
-                sb.append(", free_space [")
-                    .append(fsPath.getFree())
-                    .append("], usable_space [")
-                    .append(fsPath.getAvailable())
-                    .append("], total_space [")
-                    .append(fsPath.getTotal())
-                    .append("], mount [")
-                    .append(fsPath.getMount())
-                    .append("], type [")
-                    .append(fsPath.getType())
-                    .append(']');
-            }
-            logger.debug("node data locations details:{}", sb);
-        } else if (logger.isInfoEnabled()) {
-            FsInfo.Path totFSPath = new FsInfo.Path();
-            Set<String> allTypes = new HashSet<>();
-            Set<String> allMounts = new HashSet<>();
-            for (NodePath nodePath : nodePaths) {
-                FsInfo.Path fsPath = FsProbe.getFSInfo(nodePath);
-                String mount = fsPath.getMount();
-                if (allMounts.contains(mount) == false) {
-                    allMounts.add(mount);
-                    String type = fsPath.getType();
-                    if (type != null) {
-                        allTypes.add(type);
-                    }
-                    totFSPath.add(fsPath);
-                }
-            }
+        if (logger.isInfoEnabled()) {
```

Comment:
I don't think we should change the log messages here. Can you put it back to the debug/info differentiation, it's only that there's one path to log in each case now.

Reply:
The difference between the two cases was now only a slight difference in format, and the debug case had free space as well. This didn't seem worth a differentiation. Can you elaborate on why we should keep two slightly different log messages for the exact same info?

Reply:
Sorry I misspoke, one other difference was whether the actual path was printed.

Reply:
Mainly because changing the log messages is orthogonal to removing support for multiple data paths in `NodeEnvironment`.

Reply:
Fair enough. I pushed 81ba52c00e5

Reply:
I also pushed d914d6ac69c because the plural language of the message no longer makes sense.